tools/modules/update_payload_cached_sizes.rb

```diff
@@ -27,10 +27,16 @@
 framework = Msf::Simple::Framework.create('DisableDatabase' => true)
 
 framework.payloads.each_module do |name, mod|
-  next if name =~ /generic/
-  mod_inst = framework.payloads.create(name)
-  #mod_inst.datastore.merge!(framework.datastore)
-  next if Msf::Util::PayloadCachedSize.is_cached_size_accurate?(mod_inst)
-  $stdout.puts "[*] Updating the CacheSize for #{mod.file_path}..."
-  Msf::Util::PayloadCachedSize.update_module_cached_size(mod_inst)
-end
+  begin
+    next if name =~ /generic/
+    mod_inst = framework.payloads.create(name)
+    #mod_inst.datastore.merge!(framework.datastore)
+    next if Msf::Util::PayloadCachedSize.is_cached_size_accurate?(mod_inst)
+    $stdout.puts "[*] Updating the CacheSize for #{mod.file_path}..."
+    Msf::Util::PayloadCachedSize.update_module_cached_size(mod_inst)
+  rescue => e
+    print_line("Caught Error while updating #{name}:\n#{e}")
```

Comment:
We could report an error code on exit and still be able to process all viable payloads.  This would allow adding the testing job as well.

Consider holding error data in an array and printing as a final summary.

Reply:
I was thinking this over, and there are two ways I could see us doing this.  After we log and print the exceptions, we could call exit(1) at the end if we encountered an exception, or we could raise the first exception in the array to more closely match the error.
Thoughts?

Reply:
> exit(1) at the end if we encountered an exception

Seems like enough since the error is already logged.

Reply:
Done